e2e/tests/upgrades/upgrade_test.go

```diff
@@ -80,14 +86,17 @@ func (s *UpgradeTestSuite) UpgradeChain(ctx context.Context, chain *cosmos.Cosmo
 		s.ExecuteAndPassGovV1Beta1Proposal(ctx, chain, wallet, upgradeProposal)
 	}
 
-	height, err := chain.Height(ctx)
-	s.Require().NoError(err, "error fetching height before upgrade")
-
 	timeoutCtx, timeoutCtxCancel := context.WithTimeout(ctx, time.Minute*2)
 	defer timeoutCtxCancel()
 
-	err = test.WaitForBlocks(timeoutCtx, int(haltHeight-height)+1, chain)
-	s.Require().Error(err, "chain did not halt at halt height")
+	err = test.WaitForCondition(time.Minute*2, time.Second*2, func() (bool, error) {
+		status, err := chain.GetNode().Client.Status(timeoutCtx)
```

Comment:
unsure if we need to pass in the `timeoutCtx` to the `Status` call, if I understand it correctly, it means each status call will have 2 minutes to execute but the `WaitForCondition` fn constructs a context based on the polling interval and poll period provided.

Don't think it actually matters, but I think this could just be a `context.TODO()` instead.

Reply:
Oh yeah, you are right. It was left over from the previous version there, it is not necessary anymore.